Feature/ 업로드된 파일 삭제 (soft delete)

dyahxhx opened the pull request:
## 📄 작업 내용 요약
- 게시글에 업로드된 파일을 삭제하는 API 구현
   - Soft Delete 방식으로 삭제된 파일 이력 관리
   - 작성자 권한 검증을 통한 안전한 파일 삭제
  
<img width="1022" height="384" alt="image" src="https://github.com/user-attachments/assets/7814e4d7-c73c-4668-93ef-5766f6ce5cb8" />

<img width="1624" height="1414" alt="image" src="https://github.com/user-attachments/assets/4cea010b-b409-4599-9a15-9b883159222c" />

### 검증 로직
   1. ✅ 프로젝트 존재 여부 확인
   2. ✅ 게시글 존재 여부 확인
   3. ✅ 게시글이 프로젝트에 속하는지 검증
   4. ✅ 작성자 권한 검증 (본인만 삭제 가능)
   5. ✅ 파일이 게시글에 속하는지 검증


## 📎 Issue 76

<!-- closed #76 -->

eastVirusDa committed:
develop pull할겁니다

# Conflicts:
#	src/main/java/org/etmetmy/bn_server/domain/file/dto/FileCreateRequest.java
#	src/main/java/org/etmetmy/bn_server/domain/link/entity/Link.java

dyahxhx committed:
feat: 업로드된 파일 삭제(soft delete)

dyahxhx committed:
Merge branch 'develop' into feature/#76-post-file-delete

dyahxhx committed:
Fix comment formatting in validateWriter method

eastVirusDa commented on src/main/java/org/etmetmy/bn_server/domain/file/service/FileServiceImpl.java:
```diff
@@ -47,6 +56,44 @@ public List<ActiveFileListDTO> findAllByProjectId(Long projectId, HttpSession se
+        fileRepository.save(file);
+
+        // 삭제된 파일 목록 반환
+        return fileRepository.findAllByPostAndIsDeletedTrue(post)
```
이부분을 삭제했으면 좋겠습니다.

eastVirusDa replied:
> 이부분을 삭제했으면 좋겠습니다.

왜냐하면 이 기능은 delete만 하는 거잖아요
반환은 안해도 될 것 같습니다.
그래도 위의 Converter의 from은 수정해주세요 ^^

eastVirusDa commented on src/main/java/org/etmetmy/bn_server/domain/file/service/FileServiceImpl.java:
```diff
@@ -47,6 +56,44 @@ public List<ActiveFileListDTO> findAllByProjectId(Long projectId, HttpSession se
+        // 권한 체크 (작성자만 가능)
+        PostServiceImpl.validateWriter(post, loginUserId);
+
+        //파일 검증
```
주석 형식 맞춰주세요 :)
ex) // 파일 검증

eastVirusDa commented on src/main/java/org/etmetmy/bn_server/domain/file/controller/FileController.java:
```diff
@@ -28,4 +31,18 @@ public CommonResponse<List<ActiveFileListDTO>> getFiles(
+    @DeleteMapping("/posts/{postId}/files/{fileId}")
+    public CommonResponse<List<FileTrashResponse>> deletePostFiles(
+            @PathVariable Long projectId,
+            @PathVariable Long postId,
```
어떻게 보면 단순히 file의 is_deleted의 컬럼 하나만 바꾸는건데
projectId, postId이 필요한가 의문이 드네요
더 보니깐 검증하려고 넣은거 같은데 굳이굳이 필요한지 한번 더 생각보시는 추천드립니다.

eastVirusDa commented on src/main/java/org/etmetmy/bn_server/domain/file/service/FileServiceImpl.java:
```diff
@@ -47,6 +56,44 @@ public List<ActiveFileListDTO> findAllByProjectId(Long projectId, HttpSession se
+
+        // 권한 체크 (작성자만 가능)
+        PostServiceImpl.validateWriter(post, loginUserId);
+
```
User user = userRepository.findById(loginUserId)
                .orElseThrow(UserNotFoundException::new);
로 user를 찾으면 해당 user의 Id와 Name을 찾을 수 있습니다.
다만 이렇게 하시면 Converter에 약간의 수정이 필요합니다. 제가 추천해드릴게요

eastVirusDa commented on src/main/java/org/etmetmy/bn_server/domain/file/dto/response/FileTrashResponse.java:
```diff
@@ -0,0 +1,48 @@
+
+
+    public static class Converter {
+        public static FileTrashResponse from(File file) {
```
해당 메서드 매개변수에 User user를 추가하시면 아래 uploadUserId, uploadUserName을 쉽게 찾을 수 있습니다.

dyahxhx committed:
PR 피드백 반영/ 반환타입 삭제 및 주석 수정

# Conflicts:
#	src/main/java/org/etmetmy/bn_server/domain/file/repository/FileRepository.java

dyahxhx committed:
Merge branch 'develop' into feature/#76-post-file-delete

dyahxhx committed:
Merge branch 'develop' into feature/#76-post-file-delete

eastVirusDa reviewed (approved):
LGTM :)
